src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs

```diff
@@ -208,10 +216,15 @@ private long GetOptions(SslProtocols protocols)
                 }
                 else
                 {
-                    inputHandle = GCHandle.Alloc(inputBuffer.token, GCHandleType.Pinned);
-                    inputPtr = Marshal.UnsafeAddrOfPinnedArrayElement(inputBuffer.token, inputBuffer.offset);
-                    done = Interop.OpenSsl.DoSslHandshake(context.DangerousGetHandle(), inputPtr, inputBuffer.size, out outputPtr, out outputSize);
-                }           
+                    unsafe
+                    {
+                        fixed (byte* tokenPtr = inputBuffer.token)
+                        {
+                            inputPtr = new IntPtr(tokenPtr + inputBuffer.offset);
+                            done = Interop.OpenSsl.DoSslHandshake(context.DangerousGetHandle(), inputPtr, inputBuffer.size, out outputPtr, out outputSize);
```

Comment:
Missing //TODO or clarifying comment here? As @stephentoub mentioned in the older PR: you're using `DangerousGetHandle()` w/o any life-time management for that handle.
What is the guarantee that the context handle is the one that you care about.

In my opinion, until proven otherwise this is a _security_ risk.